src/components/icon/icon.pw.tsx

```diff
@@ -0,0 +1,538 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+import { IconProps } from "components/icon";
+import { IconComponent, IconTooltipComponent } from "./component.test-pw";
+import { getDataElementByValue } from "../../../playwright/components/index";
+import { SIZE, COLOR, CHARACTERS } from "../../../playwright/support/constants";
+
+import { checkAccessibility } from "../../../playwright/support/helper";
+
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+const colorData = [COLOR.ORANGE, COLOR.RED, COLOR.BLACK, COLOR.BROWN];
+
+test.describe("should check Icon component properties", () => {
+  test(`should check when aria-hidden prop is set as true`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<IconComponent aria-hidden />);
+    const iconLocator = await page.getByTestId("icon");
+    await expect(iconLocator).toHaveAttribute("aria-hidden", "true");
+  });
+
+  test("should check when aria-hidden prop is set as false", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<IconComponent aria-hidden={false} />);
+    const iconLocator = await page.getByTestId("icon");
+    await expect(iconLocator).toHaveAttribute("aria-hidden", "false");
+  });
+
+  testData.forEach((ariaLabel) => {
+    test(`should check when ariaLabel prop is set as ${ariaLabel}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconComponent ariaLabel={ariaLabel} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await expect(iconLocator).toHaveAttribute("aria-label", ariaLabel);
+    });
+  });
+
+  testData.forEach((id) => {
+    test(`should check icon id as ${id}`, async ({ mount, page }) => {
+      await mount(<IconComponent id={id} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await expect(iconLocator).toHaveId(id);
+    });
+  });
+
+  testData.forEach((tooltipMessage) => {
+    test(`should check tooltipMessage as ${tooltipMessage}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconTooltipComponent tooltipMessage={tooltipMessage} />);
+
+      const iconLocator = await page.getByTestId("icon");
```

Comment:
await is not necessary here. This will work: 
`const iconLocator = page.getByTestId("icon");`